planner/core/exhaust_physical_plans.go

```diff
@@ -448,7 +420,10 @@ func (p *LogicalJoin) getIndexJoinByOuterIdx(prop *property.PhysicalProperty, ou
 			continue
 		}
 		indexInfo := path.index
-		err := helper.analyzeLookUpFilters(indexInfo, ds, innerJoinKeys)
+		emptyRange, err := helper.analyzeLookUpFilters(indexInfo, ds, innerJoinKeys)
+		if emptyRange {
```

Comment:
Otherwise we may need to return a `HashJoin{Reader, Dual}`. Which i think is a little strange.

Reply:
Can you add a unit test to cover it?

Reply:
added